packages/sign_in_with_apple/lib/src/sign_in_with_apple.dart

```diff
@@ -183,15 +179,15 @@ class SignInWithApple {
   /// - Android
   ///
   /// In case Sign in with Apple is not available, the returned Future completes with `false`.
-  static Future<bool> isAvailable() {
-    return channel.invokeMethod<bool>('isAvailable');
+  static Future<bool> isAvailable() async {
+    return (await channel.invokeMethod<bool>('isAvailable')) ?? false;
   }
 
   static Future<AuthorizationCredentialAppleID> _signInWithAppleAndroid({
-    @required List<AppleIDAuthorizationScopes> scopes,
-    @required WebAuthenticationOptions webAuthenticationOptions,
-    @required String nonce,
-    @required String state,
+    @required required List<AppleIDAuthorizationScopes> scopes,
+    @required required WebAuthenticationOptions webAuthenticationOptions,
+    @required String? nonce,
+    @required String? state,
```

Comment:
It is true that there is no need to have `@required required`, but there shouldn't actually be _any_ `@required` in code with null safety, because `required` is checked consistently as part of the type system.

Line 189 and 190 should most likely be changed to use `required`, and it may then be possible to keep using the type `String`. (A nullable type for an optional named parameter is possible, but it is most likely a mistake.)

Reply:
> A nullable type for a required named parameter is possible, but it is often a mistake.

In this case it's explicitly desired, as per our preferred coding style we want the caller to pass all parameters, even if they only have `null`.